src/classes/EP_ManageEPTemplate_CTRL.cls

```diff
@@ -277,6 +277,10 @@ public with sharing class EP_ManageEPTemplate_CTRL {
             for (TaskWrapper wrapper : mapTaskWrappers.values()) {
                 Engagement_Plan_Task__c epTask = wrapper.detail;
                 epTask.Engagement_Plan_Template__c = template.id;
+                if (String.isBlank(epTask.Name)) {
+                    epTask.Name.addError(Label.NameRequired);
```

Comment:
don't you need to throw an exception in order for it to get displayed on the page.  I thought AddError only worked inside a trigger trying to update or save an object.  But since you return null after the addError, I didn't think it would cause the error to get added to pageMessages.